### ISSUE TITLE
asm.cond doesn't evaluate expressions

### ISSUE THREAD

**Issue by tmth-iskhakov** [ohler55/ojg#138]
`cond` function description says "All arguments must be array of two elements. The first element must **evaluate** to a boolean", but it doesn't evaluate provided expression. We can take a line from `cond_test`:
```
..
[set $.asm.a [cond [true abc]]]
..
```
and replace it with
```
..
[set $.asm.a [cond [[not false] abc]]]
..
```
test will fail because `not` expression will not be evaluated (can be checked in debbuger)

**Comment by tmth-iskhakov** [comment]
"local variable" workaround can be used. Instead of 
```
...
[set $.asm.a 
   [cond 
      [long_logical_expression value_on_true]
   ]
]
```
the following plan can be used:
```
[set $.local.check long_logical_expression]
[set $.asm.a 
   [cond 
      ['$.local.check' value_on_true]
   ]
]
```

**Comment by ohler55** [comment]
I'll update the tests and get it fixed. Travelling now so it might take a few days.

**Comment by ohler55** [comment]
Please try the "cold-eval" branch. It needs a few more tests but seems to work so far.

**Comment by tmth-iskhakov** [comment]
It works now. Thanks a lot.

**Comment by ohler55** [comment]
Released v1.19.3 with the fix.
